src/nvim/option.c

```diff
@@ -3212,63 +3212,67 @@ static void set_option_varp(OptIndex opt_idx, void *varp, OptVal value, bool fre
   UNREACHABLE;
 }
 
-/// Return C-string representation of OptVal. Caller must free the returned C-string.
-static char *optval_to_cstr(OptVal o)
+/// Get C-string representation of OptVal. Uses vim.inspect() under the hood.
+/// Always returns a valid C-string.
+///
+/// @param  value  Option value.
+///
+/// @return  C-string representation of value.
+static const char *optval_to_cstr(OptVal value)
 {
-  switch (o.type) {
-  case kOptValTypeNil:
-    return xstrdup("");
-  case kOptValTypeBoolean:
-    return xstrdup(o.data.boolean ? "true" : "false");
-  case kOptValTypeNumber: {
-    char *buf = xmalloc(NUMBUFLEN);
-    snprintf(buf, NUMBUFLEN, "%" PRId64, o.data.number);
-    return buf;
-  }
-  case kOptValTypeString: {
-    char *buf = xmalloc(o.data.string.size + 3);
-    snprintf(buf, o.data.string.size + 3, "\"%s\"", o.data.string.data);
-    return buf;
-  }
+  Error err = ERROR_INIT;
+  const char *res = nlua_inspect(optval_as_object(value), &err);
```

Comment:
is there a reason you can't use `NLUA_EXEC_STATIC` instead of adding `nlua_inspect` ?